tests/migration_test.go

```diff
@@ -345,35 +346,39 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 	}
 	confirmVMIPostMigrationAborted := func(vmi *v1.VirtualMachineInstance, migrationUID string, timeout int) *v1.VirtualMachineInstance {
 		By("Waiting until the migration is completed")
-		Eventually(func() bool {
+		EventuallyWithOffset(1, func() v1.VirtualMachineInstanceMigrationState {
 			vmi, err = virtClient.VirtualMachineInstance(vmi.Namespace).Get(vmi.Name, &metav1.GetOptions{})
-			Expect(err).ToNot(HaveOccurred())
+			ExpectWithOffset(2, err).ToNot(HaveOccurred())
 
-			if vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Completed &&
-				vmi.Status.MigrationState.AbortStatus == v1.MigrationAbortSucceeded {
-				return true
+			if vmi.Status.MigrationState != nil {
+				return *vmi.Status.MigrationState
 			}
-			return false
+			return v1.VirtualMachineInstanceMigrationState{}
```

Comment:
Here is an alternative that does not require the reader to interpret this `gstruct` new thing.

```
var migrationState v1.VirtualMachineInstanceMigrationState
if vmi.Status.MigrationState != nil {
    migrationState.Completed = vmi.Status.MigrationState.Completed
    migrationState.AbortStatus = vmi.Status.MigrationState.AbortStatus
}

return migrationState
```

Then equal against `v1.VirtualMachineInstanceMigrationState{Completed: true, AbortStatus: v1.MigrationAbortSucceeded}`

Reply:
Using equal produces
```
 Timed out after 180.001s.
  Expected
      <v1.VirtualMachineInstanceMigrationState>: {
          StartTimestamp: nil,
          EndTimestamp: nil,
          TargetNodeDomainDetected: false,
          TargetNodeAddress: "",
          TargetDirectMigrationNodePorts: nil,
          TargetNode: "",
          TargetPod: "",
          TargetAttachmentPodUID: "",
          SourceNode: "",
          Completed: true,
          Failed: false,
          AbortRequested: false,
          AbortStatus: "Succeeded",
          MigrationUID: "",
          Mode: "",
          MigrationPolicyName: nil,
          MigrationConfiguration: nil,
          TargetCPUSet: nil,
          TargetNodeTopology: "",
      }
  to equal
      <v1.VirtualMachineInstanceMigrationState>: {
          StartTimestamp: nil,
          EndTimestamp: nil,
          TargetNodeDomainDetected: false,
          TargetNodeAddress: "",
          TargetDirectMigrationNodePorts: nil,
          TargetNode: "",
          TargetPod: "",
          TargetAttachmentPodUID: "",
          SourceNode: "",
          Completed: false,
          Failed: false,
          AbortRequested: false,
          AbortStatus: "Succeeded",
          MigrationUID: "",
          Mode: "",
          MigrationPolicyName: nil,
          MigrationConfiguration: nil,
          TargetCPUSet: nil,
          TargetNodeTopology: "",

```
on the other had `gstruct`

```
Timed out after 180.000s.
  Expected
      <string>: VirtualMachineInstanceMigrationState
  to match fields: {
  .Completed:
  	Expected
  	    <bool>: true
  	to be false
  }
```

I found the latter to be more clear and more valuable. WDYT?

Reply:
Yes, it has a better/clearer report but costs more to understand the code.
I leave it up to you, there is no clear answer what is better.